pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -50,12 +50,38 @@ func ValidateMachinePool(p *openstack.MachinePool, ci *CloudInfo, controlPlane b
 		}
 	}
 
+	allErrs = append(allErrs, validateZones(p.Zones, ci.Zones, fldPath.Child("zones"))...)
 	allErrs = append(allErrs, validateUUIDV4s(p.AdditionalNetworkIDs, fldPath.Child("additionalNetworkIDs"))...)
 	allErrs = append(allErrs, validateUUIDV4s(p.AdditionalSecurityGroupIDs, fldPath.Child("additionalSecurityGroupIDs"))...)
 
 	return allErrs
 }
 
+func validateZones(input []string, actual []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Base Case: No user input
+	if len(input) == 1 {
+		if input[0] == "" {
+			return allErrs
+		}
+	}
```

Comment:
what is this bad case?? wouldn't this be handles already when users provide an element to the array that does not belong to available...

Reply:
Its not, its just confusing. It's returning an empty error list